pkg/cmd/render/render.go

```diff
@@ -0,0 +1,165 @@
+func (ppf *performanceProfileFiles) String() string {
+	return fmt.Sprint(*ppf)
+}
+
+func (ppf *performanceProfileFiles) Type() string {
+	return "performanceProfileFiles"
+}
+
+// Set parse performance-profile-input-files flag and store it in ppf
+func (ppf *performanceProfileFiles) Set(value string) error {
+	if len(*ppf) > 0 {
+		return errors.New("performance-profile-input-files flag already set")
+	}
+
+	for _, s := range strings.Split(value, ",") {
+		*ppf = append(*ppf, s)
+	}
+	return nil
+}
+
+//NewRenderCommand creates a render command.
+func NewRenderCommand() *cobra.Command {
+	renderOpts := renderOpts{}
+
+	cmd := &cobra.Command{
+		Use:   "render",
+		Short: "Render performance-addon-operator manifests",
+		Run: func(cmd *cobra.Command, args []string) {
+
+			if err := renderOpts.Validate(); err != nil {
+				klog.Fatal(err)
+			}
+
+			if err := renderOpts.Run(); err != nil {
+				klog.Fatal(err)
+			}
+		},
+	}
+
+	renderOpts.AddFlags(cmd.Flags())
+
+	return cmd
+}
+
+func (r *renderOpts) AddFlags(fs *pflag.FlagSet) {
+	fs.Var(&r.performanceProfileInputFiles, "performance-profile-input-files", "A comma-separated list of performance-profile manifests.")
+	fs.StringVar(&r.assetsInDir, "asset-input-dir", components.AssetsDir, "Input path for the assets directory.")
+	fs.StringVar(&r.assetsOutDir, "asset-output-dir", r.assetsOutDir, "Output path for the rendered manifests.")
+	// environment variables has precedence over standard input
+	r.readFlagsFromEnv()
+}
+
+func (r *renderOpts) readFlagsFromEnv() {
+	if ppInFiles := os.Getenv("PERFORMANCE_PROFILE_INPUT_FILES"); len(ppInFiles) > 0 {
+		r.performanceProfileInputFiles.Set(ppInFiles)
+	}
+
+	if assetInDir := os.Getenv("ASSET_INPUT_DIR"); len(assetInDir) > 0 {
+		r.assetsInDir = assetInDir
+	}
+
+	if assetsOutDir := os.Getenv("ASSET_OUTPUT_DIR"); len(assetsOutDir) > 0 {
+		r.assetsOutDir = assetsOutDir
+	}
+}
+
+func (r *renderOpts) Validate() error {
+	if len(r.performanceProfileInputFiles) == 0 {
+		return fmt.Errorf("performance-profile-input-files must be specified")
+	}
+
+	if len(r.assetsOutDir) == 0 {
+		return fmt.Errorf("asset-output-dir must be specified")
+	}
+
+	return nil
+}
+
+func (r *renderOpts) Run() error {
+	for idx, pp := range r.performanceProfileInputFiles {
+		b, err := ioutil.ReadFile(pp)
+		if err != nil {
+			return err
+		}
+
+		profile := &performancev2.PerformanceProfile{}
+		err = yaml.Unmarshal(b, profile)
+		if err != nil {
+			return err
+		}
+
+		components, err := manifestset.GetNewComponents(profile, &r.assetsInDir)
+		if err != nil {
+			return err
+		}
+		or := []v1.OwnerReference{
+			{
+				Kind: profile.Kind,
+				Name: profile.Name,
+			},
+		}
+
+		for _, componentObj := range components.ToObject() {
+			componentObj.SetOwnerReferences(or)
+		}
+
+		for name, manifest := range components.ToManifestTable() {
+			b, err := yaml.Marshal(manifest)
+			if err != nil {
+				return err
+			}
+
+			fileName := fmt.Sprintf("%s-%d.yaml", name, idx)
```

Comment:
ok, this is the biggest and last point. I believe the output name should be like `$INPUT_NAME_tuned.yaml` `$INPUT_NAME_kubeletconfig.yaml` and so forth. IOW, use kind (`in_this_format`, `NotLikeThis`) and not a index. In case of ambiguity, we shoudl differentiate with object name if possible